lib/cylc/job.sh

```diff
@@ -36,26 +36,15 @@ cylc__job__main() {
         set -x
     fi
     # Prelude
-    typeset file_name=
+    typeset JOB_INIT_ENV=
     # conf/job-init-env.sh for back-compat pre 7.7.0.
-    for file_name in \
-        "${HOME}/.cylc/job-init-env.sh" \
-        "${CYLC_DIR}/etc/job-init-env.sh" \
-        "${CYLC_DIR}/conf/job-init-env.sh"
-    do
-        if [[ -f "${file_name}" ]]; then
-            if "${CYLC_DEBUG:-false}"; then
-                . "${file_name}"
-            else
-                . "${file_name}" 1>'/dev/null' 2>&1
-            fi
-            break
+    JOB_INIT_ENV="${HOME}/.cylc/job-init-env.sh"
+    if [[ -f "${JOB_INIT_ENV}" ]]; then
+        if "${CYLC_DEBUG:-false}"; then
+            . "${JOB_INIT_ENV}"
+        else
+            . "${JOB_INIT_ENV}" 1>'/dev/null' 2>&1
```

Comment:
I would argue that we should get rid of this logic altogether - now that we have `bash -l` in the job script header.

Reply:
Passed over my head. I can dig deeper to see if I understand why `bash -l` would replace it... but I trust another reviewer will have better understanding to comment :grimacing: (and promise to lurk and try to learn a bit more Shell)

Reply:
In the past, we would call `bash` without the `--login` (i.e. `-l`). So users need to set up their environment for the job somehow (so it has access to Cylc, etc). Now that the job script calls `bash` with `--login` (i.e. `-l`), we can safely assume that the correct environment is already available as `--login` will invoke the site/user login profiles.

Reply:
We still have the option to not use a login shell, however (security people seem not to like jobs using login shells...) ... in which case there might still be a requirement for cylc-specific setup??

Reply:
Like to find out the reason why security people are not keen on login shell. After all, users can put login profiles to the script that get sourced here. If anything, the login shell is a lot more standard and can be more secure, if done right.

Reply:
I know that at least in some Node.js and PHP apps, it is common to have keys and tokens as environment variables. Some times under global envvars. I think Django may use that as well?

So maybe some people are worried about those variables being written to log files or to the web interface, leaking the credentials... Just guessing.

Reply:
That would be a bad setup. The user would have to have deliberately written the logic to print out the tokens or have xtrace switched on in the wrong place.

They can do all these stuffs in this script as well.

Reply:
A `printenv` command later can also do it.

Reply:
I agree. But if instead they disallowed `bash -l`, I think it would not have happened... which means lower risk, from a security point of view.

My preference is for having separate user accounts, possibly even separated boxes (VM, container, etc), and instead of worrying too much about it, simply follow what is in the user manual. So this could be a possible answer to users concerned about security... though not all would accept I suspect...